docs/spec/spec.md

```diff
@@ -0,0 +1,556 @@
+# Specification: Ballerina FTP Library
+
+_Owners_: @shafreenAnfar @dilanSachi @Bhashinee    
+_Reviewers_: @shafreenAnfar @Bhashinee  
+_Created_: 2020/10/28   
+_Updated_: 2022/02/07  
+_Edition_: Swan Lake    
+_Issue_: [#2202](https://github.com/ballerina-platform/ballerina-standard-library/issues/2202)
+
+# Introduction
+This is the specification for the FTP standard library of [Ballerina language](https://ballerina.io/), which provides FTP client/listener functionalities to send and receive files by connecting to a FTP/SFTP server.
+
+The FTP library specification has evolved and may continue to evolve in the future. Released versions of the specification can be found under the relevant Github tag.
+
+If you have any feedback or suggestions about the library, start a discussion via a Github issue or in the [Slack channel](https://ballerina.io/community/). Based on the outcome specification and implementation can be updated. Community feedback is always welcome. Any accepted proposal which affects the specification is stored under `/docs/proposals`. Proposals under discussion can be found with the label `type/proposal` in Github.
+
+Conforming implementation of the specification is released and included in the distribution. Any deviation from the specification is considered a bug.
+
+# Contents
+1. [Overview](#1-overview)
+2. [Configurations](#2-configurations)
+    *  2.1. [Security Configurations](#21-security-configurations)
+    *  2.2. [FileInfo](#22-fileinfo)
+3. [Client](#3-client)
+    *  3.1. [Configurations](#31-configurations)
+    *  3.2. [Initialization](#32-initialization)
+        *    3.2.1. [Insecure Client](#321-insecure-client)
+        *    3.2.2. [Secure Client](#322-secure-client)
+    *  3.3. [Functions](#33-functions)
+4. [Listener](#4-listener)
+    *  4.1. [Configurations](#41-configurations)
+    *  4.2. [Initialization](#42-initialization)
+        *  4.2.1. [Insecure Listener](#421-insecure-listener)
+        *  4.2.2. [Secure Listener](#422-secure-listener)
+    *   4.3. [Usage](#43-usage)
+5. [Samples](#5-samples)
+    *  5.1. [Sending Files](#51-sending-files)
+    *  5.2. [Listening to File Changes](#52-listening-to-file-changes)
+
+## 1. Overview
+FTP is the traditional file transfer protocol. It’s a basic way of using the Internet to share files.
+SFTP (or Secure File Transfer Protocol) is an alternative to FTP that also allows transferring files,
+but adds a layer of security to the process. SFTP uses SSH (or secure shell) encryption to protect data as
+it’s being transferred. This means data is not exposed to outside entities on the Internet when it is sent
+to another party.
```

Comment:
```suggestion
to another party. This library provides support for both protocols. 
```

Reply:
Updated in https://github.com/ballerina-platform/module-ballerina-ftp/pull/538/commits/3df62c276651402b0d11a31cc5968282ed4cba5f